src/qz/printer/status/StatusMonitor.java

```diff
@@ -204,9 +207,10 @@ private synchronized static boolean dispatchStatusEvent() {
             // And find every client that subscribed to all printers
             listeningConnections.addAll(clientPrinterConnections.get(ALL_PRINTERS));
         }
-        for (SocketConnection connection : listeningConnections) {
-            statusSessions.get(connection).statusChanged(status);
-        }
+
+        for(SocketConnection connection : listeningConnections)
+            statusSessions.get(connection).statusChanged(status, () -> listeningConnections.remove(connection));
```

Comment:
`listeningConnections` is a local hash built each call, so shouldn't this be removing the connection from the source list under the `clientPrinterConnections` instead?

And minor nitpick, but can we add the brackets back to this loop?

Reply:
> And minor nitpick, but can we add the brackets back to this loop?

Whoops.... I thought I was helping the diff by removing them but I clearly misread the original code.

With regards to the variable scope, I think it's identical to your commit 1183887363bbe92b42adddc64b40ed8b22ca2766 so I kept the same scope.  I'll study the code to find out what it's actually doing....

Reply:
I think this can be resolved, please mark it if so.  I'll also ping @Vzor- for a second set of eyes.

Reply:
If the intention is to stop listening to the specific printer, `clientPrinterConnections` is where that should be added. https://github.com/qzind/tray/commit/326f2960b6eb6c78deeac5966cfeedaedad73faa will not have the intended effect, it will simply throw a NPE during the next status. I will write a quick example of what it should look like.

Reply:
> If the intention is to stop listening to the specific printer, clientPrinterConnections is where that should be done. https://github.com/qzind/tray/commit/326f2960b6eb6c78deeac5966cfeedaedad73faa will not have the intended effect, it will simply throw a NPE during the next status. I will write a quick example of what it should look like.

It's not the intention.  The intention is to stop attempting to send data to that websocket.

Reply:
Done via bde79b5.